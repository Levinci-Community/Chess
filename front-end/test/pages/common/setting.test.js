import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import SettingsPage from "../../../src/pages/common/settings";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (key) => key,
  }),
}));

test("renders SettingsPage snapshot", () => {
  const { asFragment } = render(<SettingsPage />);
  expect(asFragment()).toMatchSnapshot();
});
