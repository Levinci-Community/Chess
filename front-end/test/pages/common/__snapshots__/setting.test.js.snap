// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SettingsPage snapshot 1`] = `
<DocumentFragment>
  <div
    class="chakra-container css-1c3tr3p"
  >
    <h2
      class="chakra-heading css-1qm1lh"
    >
      settings.heading
    </h2>
    <div
      class="css-ohco1m"
    >
      <div
        class="chakra-stack css-f8f1y1"
      >
        <h2
          class="chakra-heading css-wlea3r"
        >
          settings.languageSettings
        </h2>
        <div
          class="css-17xejub"
        />
        <div
          class="chakra-select__wrapper css-wcqg5i"
        >
          <select
            class="chakra-select css-1ik61og"
          >
            <option
              value="en"
            >
              settings.english
            </option>
            <option
              value="vi"
            >
              settings.vietnamese
            </option>
          </select>
          <div
            class="chakra-select__icon-wrapper css-162mkon"
          >
            <svg
              aria-hidden="true"
              class="chakra-select__icon"
              focusable="false"
              role="presentation"
              style="width: 1em; height: 1em; color: currentColor;"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-ohco1m"
    >
      <div
        class="chakra-stack css-f8f1y1"
      >
        <h2
          class="chakra-heading css-wlea3r"
        >
          settings.themeSettings
        </h2>
        <div
          class="css-17xejub"
        />
        <div
          class="chakra-select__wrapper css-wcqg5i"
        >
          <select
            class="chakra-select css-1ik61og"
          >
            <option
              value="light"
            >
              settings.light
            </option>
            <option
              value="dark"
            >
              settings.dark
            </option>
          </select>
          <div
            class="chakra-select__icon-wrapper css-162mkon"
          >
            <svg
              aria-hidden="true"
              class="chakra-select__icon"
              focusable="false"
              role="presentation"
              style="width: 1em; height: 1em; color: currentColor;"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
